express-delete/index.js

```diff
@@ -0,0 +1,45 @@
+const express = require('express');
+const app = express();
+
+const grades = {
+  12: {
+    id: 12,
+    name: 'Tim Berners-Lee',
+    course: 'HTML',
+    score: 95
+  },
+  47: {
+    id: 47,
+    name: 'Brendan Eich',
+    course: 'JavaScript',
+    score: 100
+  },
+  273: {
+    id: 273,
+    name: 'Forbes Lindsay',
+    course: 'JavaScript',
+    score: 92
+  }
+};
+
+app.get('/api/grades', (req, res) => {
+  const gradesArray = [];
+  for (const id in grades) {
+    gradesArray.push(grades[id]);
+  }
+  res.json(gradesArray);
+});
+
+app.delete('/api/grades/:id', (req, res) => {
+  for (const id in grades) {
+    if (req.params.id === id) {
+      delete grades[id];
+      res.sendStatus(204);
+    }
+  }
```

Comment:
When you already know the property name you are looking for, in this case the `id`, you don't need a loop. You can do:
```JAVASCRIPT
const id = req.params.id;
delete grades[id];

res.sendStatus(204);
```